### PR TITLE
add date type to timing function

### LINKED ISSUE #168: client.timing could accept date
currently, the timing method just accepts timestamp which is time in milliseconds, which could be more convenient if a date is also accepted.

Will raise a PR

### PR DESCRIPTION
**Pull request by Impeekay**
tests needed.

fixes #168

### CODE REVIEW

**Comment by bdeitte** [comment]
Thanks for the addition.  I would like to see a test added for this before bringing in, so that we don't break it later.  You could add a very simple one around here: https://github.com/brightcove/hot-shots/blob/master/test/statsFunctions.js#L125

And two other requests here.  Could you add "(or Date object of which the difference is calculated)" to the comment line in READMEm.md, to indicate this new functionality.  And change "number" to "number | Date" for the four type lines for this function: https://github.com/brightcove/hot-shots/blob/master/types.d.ts#L88

Thanks!

**Comment by Impeekay** [comment]
Sure, will update the PR.

File: test/statsFunctions.js [hunk 1]
```diff
@@ -122,6 +122,29 @@ describe('#statsFunctions', () => {
...
+          server = createServer(serverType, opts => {
+            statsd = createHotShotsClient(opts, clientType);
+            statsd.timing('test', new Date(Date.now() - 10));
+          });
+          server.on('metrics', metrics => {
+            assert.equal(metrics, `test:10|ms${metricsEnd}`);
+            done();
+          });
+        });
+      })
```
Comment (bdeitte) [inline test/statsFunctions.js:146]: This is missing a semi-colon (which is causing the tests to fail).  Once this is fixed, I can merge this all in, thanks!

Comment (Impeekay) [reply test/statsFunctions.js:146]: Damn, did not see that. It did pass on mine when I tested it. Also it'd be better if we put out a prettier config for ppl to use. :) 
I'll update it.

**Pull request merged**